python/hidet/graph/frontend/torch/dynamo_backends.py

```diff
@@ -41,7 +42,9 @@ def generate_executor(flow_graph: FlowGraph) -> Callable:
         if use_fp16 and use_fp16_reduction:
             ctx.set_reduce_precision('float16')
         if save_dir:
-            ctx.save_graph_instrument(save_dir)
+            generate_executor.graph_counter = getattr(generate_executor, 'graph_counter', 0) + 1
+            save_dir: Path = Path(save_dir)
+            ctx.save_graph_instrument(save_dir / "graph_{}".format(generate_executor.graph_counter))
```

Comment:
the thing with find the first non-existed directory is that if I recompile the model and use the same save_dir value, it will not overwrite the previous compiled graphs, but keeps incrementing. I didn't like this behaviour. 

If you think it is OK to disallow overwriting, then I am happy to make it find the first non-existed directory

_How do you think to implement this as a seperate function?_
* you mean make this three lines into a different function?

Reply:
> the thing with find the first non-existed directory is that if I recompile the model and use the same save_dir value, it will not overwrite the previous compiled graphs, but keeps incrementing. I didn't like this behaviour.

I see. Make sense. 

> How do you think to implement this as a seperate function?

Yes. For example, something like
```python
def _resolve_save_dir(save_dir: str) -> str:
   _resolve_save_dir.counter = _resolve_save_dir.counter = getattr(..., 'counter', 0) + 1
  return str(Path(save_dir) / "graph_{}".format(_resolve_save_dir.counter))
```